cpp/locality/BondHistogramCompute.h

```diff
@@ -115,17 +113,17 @@ class BondHistogramCompute
 
 protected:
     box::Box m_box;
-    unsigned int m_frame_counter;  //!< Number of frames calculated.
-    unsigned int m_n_points;       //!< The number of points.
-    unsigned int m_n_query_points; //!< The number of query points.
-    bool m_reduce;                 //!< Whether or not the histogram needs to be reduced.
+    unsigned int m_frame_counter {0};  //!< Number of frames calculated.
+    unsigned int m_n_points {0};       //!< The number of points.
+    unsigned int m_n_query_points {0}; //!< The number of query points.
+    bool m_reduce {true};              //!< Whether or not the histogram needs to be reduced.
```

Comment:
I mentioned this before, but what is the logic behind initializing instance variables this way?

Reply:
See the other comment.